orchagent/srv6orch.cpp

```diff
@@ -422,6 +422,175 @@ bool Srv6Orch::mySidExists(string my_sid_string)
+         * nexthop from the ASIC, and add them to the SRv6 MySID entries pending set.
+         */
+
+        SWSS_LOG_INFO("Neighbor DELETE event: %s, removing associated SRv6 SIDs",
+                        update.entry.ip_address.to_string().c_str());
+
+        for (auto it = srv6_my_sid_table_.begin(); it != srv6_my_sid_table_.end();)
+        {
+            /* Skip SIDs that are not associated with a L3 Adjacency */
+            if (it->second.endAdjString.empty())
+            {
+                ++it;
+                continue;
+            }
+
+            try
+            {
+                /* Skip SIDs that are not associated with this neighbor */
+                if (IpAddress(it->second.endAdjString) != update.entry.ip_address)
+                {
+                    ++it;
+                    continue;
+                }
+            }
+            catch (const std::invalid_argument &e)
+            {
+                /* SRv6 SID is associated with an invalid L3 Adjacency IP address, skipping */
+                ++it;
+                continue;
+            }
+
+            /*
+             * Save SID entry information to temp variables, before removing the SID.
+             * This information will be consumed used later. 
+             */
+            string my_sid_string = it->first;
+            const string dt_vrf = it->second.endVrfString;
+            const string adj = it->second.endAdjString;
+            string end_action;
+            for (auto iter = end_behavior_map.begin(); iter != end_behavior_map.end(); iter++)
+            {
+                if (iter->second == it->second.endBehavior)
+                {
+                    end_action = iter->first;
+                    break;
+                }
+            }
+
+            /* Skip SIDs with unknown SRv6 behavior */
+            if (end_action.empty())
+            {
+                ++it;
+                continue;
+            }
+
+            SWSS_LOG_INFO("Removing SID %s, action %s, vrf %s, adj %s", my_sid_string.c_str(), dt_vrf.c_str(), adj.c_str(), end_action.c_str());
+
+            /* Let's delete the SID from the ASIC */
+            unordered_map<string, MySidEntry>::iterator tmp = it;
+            ++tmp;
+            if(!deleteMysidEntry(it->first))
+            {
+                SWSS_LOG_ERROR("Failed to delete my_sid entry for sid %s", it->first.c_str());
+                ++it;
+                continue;
+            }
+            it = tmp;
+
+            SWSS_LOG_INFO("SID %s removed successfully", my_sid_string.c_str());
+
+            /*
+             * Finally, add the SID to the pending MySID entries set, so that we can re-install it 
+             * when the neighbor comes back
+             */
+            auto pending_mysid_entry = make_tuple(my_sid_string, dt_vrf, adj, end_action);
+            m_pendingSRv6MySIDEntries[NextHopKey(update.entry.ip_address.to_string())].insert(pending_mysid_entry);
```

Comment:
NextHopKey with ip_address and interface?? Check all the other places where m_pendingSRv6MySIDEntries dictionary is accessed.

Reply:
@kperumalbfn Fixed.